lib/improver/wind_direction.py

```diff
@@ -0,0 +1,237 @@
+
+        # Combine components into a complex number and return.
+        return a + 1j*b
+
+    @staticmethod
+    def complex_to_deg(complex_in):
+        """Take complex number and find angle in degrees from 360.
+           the "np.angle" function returns negative numbers when the input
+           is greater than 180. Therefore additional processing is needed
+           to ensure that the angle is between 0-359."""
+
+        angle = np.angle(complex_in, deg=True)
+
+        # If the input is an array then need to use native np.functions
+        # for speed rather than applying vector functions to the array.
+        if type(angle) is np.ndarray:
+
+            # If angle negative value - add 360 degrees.
+            angle = np.where(angle < 0, angle+360, angle)
+
+            # If angle == 360 - set to zero degrees.
+            # Due to floating point - need to round value before using
+            # equal operator.
+            round_angle = np.around(angle, 2)
+            angle = np.where(round_angle == 360, 0.0, angle)
+        else:
+            if angle < 0:
+                angle = angle + 360.0
+            if round(angle, 2) == 360.00:
+                angle = 0.0
+        return angle
+
+    @staticmethod
+    def find_r_values(complex_in):
+        """Takes input wind direction in complex values and returns array
+           containing r values using Pythagoras theorem."""
+
+        return np.sqrt(np.square(complex_in.real)+np.square(complex_in.imag))
+
+    @staticmethod
+    def calc_polar_mean_std_dev(wind_dir_complex, wind_dir_deg_mean):
+        """The averaged wind direction complex values have r values
+           between 0-1. So we need to recalculate the complex values so they
+           retain their angle but r is fixed as r=1. They can then be used to
+           find the distance between the average angle and the other
+           wind direction data points. From this information - we can calculate
+           a confidence value."""
+
+        wind_dir_complex_mean_fixr = WindDirection.deg_to_complex(
+                                                             wind_dir_deg_mean)
+
+        # Find difference in the distance between all the observed points and
+        # mean point with fixed r=1.
+        # For maths to work - the "wind_dir_complex_mean_fixr array" needs to
+        # be "tiled" so that it is the same dimension as "wind_dir_complex".
+        wind_dir_complex_mean_tile = np.tile(wind_dir_complex_mean_fixr,
+                                             (wind_dir_complex.shape[0], 1, 1))
+
+        # Calculate distance from each wind direction data point to the
+        # average point.
+        difference = wind_dir_complex - wind_dir_complex_mean_tile
+        dist_from_mean = np.sqrt(np.square(difference.real) +
+                                 np.square(difference.imag))
+
+        # Find average distance.
+        dist_from_mean_avg = np.mean(dist_from_mean, axis=0)
```

Comment:
See my comment below about assuming axes

Reply:
The problem is that if the realization axis is variable then there is a massive headache with rebroadcasting the arrays with the "tile" command. I haven't been able to work out a way of doing it in such a way that it wont need rebroadcasting.